katz/src/main/kotlin/katz/data/Coproduct.kt

```diff
@@ -22,33 +22,25 @@ typealias CoproductKind<F, G, A> = HK<CoproductFG<F, G>, A>
 
 fun <F, G, A> CoproductKind<F, G, A>.ev(): Coproduct<F, G, A> = this as Coproduct<F, G, A>
 
-data class Coproduct<F, G, A>(val run: Either<HK<F, A>, HK<G, A>>) : CoproductKind<F, G, A> {
+data class Coproduct<F, G, A>(val CF: Comonad<F>, val CG: Comonad<G>, val run: Either<HK<F, A>, HK<G, A>>) : CoproductKind<F, G, A> {
 
     class F private constructor()
 
-    companion object {
-
-        fun <F, G, A> leftc(x: HK<F, A>): Coproduct<F, G, A> =
```

Comment:
They are just syntax and we are gonna do a syntax pass over all datatypes, this were left unused once I tried to make them reified and could not use them inside the class. Also is too early for syntax and it's getting in the way of refactoring and doing big PRs that touch all datatypes like this one